components/rhs_card/rhs_card.tsx

```diff
@@ -1,74 +1,81 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import deepEqual from 'fast-deep-equal';
-import PropTypes from 'prop-types';
-import React from 'react';
+import React, {ReactNode} from 'react';
 import Scrollbars from 'react-custom-scrollbars';
 import {FormattedMessage} from 'react-intl';
 import {Link} from 'react-router-dom';
 
 import DelayedAction from 'utils/delayed_action';
-import Constants, {RHSStates} from 'utils/constants';
+import Constants from 'utils/constants';
+
 import RhsCardHeader from 'components/rhs_card_header';
 import Markdown from 'components/markdown';
 import UserProfile from 'components/user_profile';
 import PostProfilePicture from 'components/post_profile_picture';
-import * as GlobalActions from 'actions/global_actions';
 
-export function renderView(props) {
+import {Post} from '@mattermost/types/posts';
+
+import {RhsState} from 'types/store/rhs';
+import {PostPluginComponent} from 'types/store/plugins';
+
+import {emitCloseRightHandSide} from 'actions/global_actions';
+
+type Props = {
+    isMobileView: boolean;
+    selected?: Post;
+    pluginPostCardTypes?: Record<string, PostPluginComponent>;
+    previousRhsState?: RhsState;
+    enablePostUsernameOverride?: boolean;
+    teamUrl?: string;
+};
+
+type State = {
+    isScrolling: boolean;
+};
+
+export function renderView(props: Props) {
```

Comment:
I was mentioning this in an earlier review already: I think we could simplify these into a single component that handles all different variations

```
const renderScrollComponent = (props: Props, type: 'view'|'horizontal'|'vertical') => (
        <div
            {...props}
            className={`scrollbar--${type}`}
        />
);
```

this might be out of scope for this PR though.